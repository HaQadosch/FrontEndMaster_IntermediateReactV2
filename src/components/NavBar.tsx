import React from 'react';
import { Link } from '@reach/router';
import { css, jsx, keyframes } from '@emotion/core';
import { colors } from '../utils/colors';

const spin = keyframes`
  to {
    transform: rotate(360deg)
  }
`;

/** @jsx jsx */
export const NavBar: React.FC = () => {
  return (
    <header
      css={css`
        background-color: ${colors.secondary};
        padding: 15px;
      `}
    >
      <Link to='/'>Adopt me !</Link>
      <span
        css={css`
          font-size: 4rem;
          &:hover {
            text-decoration: underline;
            animation: 1s ${spin} linear infinite reverse;
          }
          display: inline-block;
          animation: 5s ${spin} linear infinite;
        `}
        role='img'
        aria-label='logo'
      >
        🐩
      </span>
    </header>
  );
};
