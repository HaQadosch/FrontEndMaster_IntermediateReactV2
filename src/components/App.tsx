import React, { useState } from 'react';
import { Router } from '@reach/router';
import './App.css';
import { SearchParams } from './SearchParams';
import { Details } from './Details';
import { ThemeContext } from './ThemeContext';
import { NavBar } from './NavBar';

export const App: React.FC = () => {
  const [color, setColor] = useState<string>('darkblue');

  return (
    <ThemeContext.Provider value={[color, setColor]}>
      <div>
        <NavBar />
        <React.StrictMode>
          <Router>
            <SearchParams path='/' />
            <Details path='/details/:id' id='' />
          </Router>
        </React.StrictMode>
      </div>
    </ThemeContext.Provider>
  );
};
