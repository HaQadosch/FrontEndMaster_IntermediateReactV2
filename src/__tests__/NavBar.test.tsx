import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'react-testing-library';
import serializer from 'jest-emotion';
import { NavBar } from '../components/NavBar';

expect.addSnapshotSerializer(serializer);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavBar />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('NavBar matches the snapshot', () => {
  const { debug, asFragment } = render(<NavBar />);
  expect(asFragment()).toMatchSnapshot();
  // debug(asFragment());
});
