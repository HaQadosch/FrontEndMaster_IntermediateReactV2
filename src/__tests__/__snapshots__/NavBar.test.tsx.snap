// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches the snapshot 1`] = `
<DocumentFragment>
  @keyframes animation-0 {
  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-0 {
  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-1 {
  background-color: #d9c148;
  padding: 15px;
}

.emotion-0 {
  font-size: 4rem;
  display: inline-block;
  -webkit-animation: 5s animation-0 linear infinite;
  animation: 5s animation-0 linear infinite;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-animation: 1s animation-0 linear infinite reverse;
  animation: 1s animation-0 linear infinite reverse;
}

<header
    class="emotion-1"
  >
    <a
      aria-current="page"
      href="/"
    >
      Adopt me !
    </a>
    <span
      aria-label="logo"
      class="emotion-0"
      role="img"
    >
      üê©
    </span>
  </header>
</DocumentFragment>
`;
